src/CMakeLists.txt

```diff
@@ -8,16 +8,16 @@ if(NOT ENABLE_UNIT_TESTS)
     list(FILTER MY_SRC EXCLUDE REGEX "/tests/")
 endif()
 
-# if mpi is not enabled, remove the uneeded source files
+# if mpi is not enabled, remove the unneeded source files
 if(NOT ENABLE_MPI)
     # duplicate the list
     set(MY_SRC_BACK ${MY_SRC})
     # exclude everything from parallel
     list(FILTER MY_SRC EXCLUDE REGEX "/parallel/")
 
-    # but include DomainDecompBase* and LoadCalc*
+    # but include DomainDecompBase*, LoadCalc* and Boundary utilities
     list(FILTER MY_SRC_BACK INCLUDE REGEX "/parallel/")
-    list(FILTER MY_SRC_BACK INCLUDE REGEX "DomainDecompBase|LoadCalc|Zonal|ForceHelper")
+    list(FILTER MY_SRC_BACK INCLUDE REGEX "boundaries/|DomainDecompBase|LoadCalc|Zonal|ForceHelper")
```

Comment:
The option to enable MaMiCo coupling in the cmake does not check for MPI. Are the boundariey code parts realy only necessary for MPI?

Reply:
No, the boundary code works without MPI. This regex is to explicitly include things in the `parallel` directory that work without MPI. I agree this is confusing, I opened #343 to document that.